### ISSUE TITLE
NullPointerException on MapboxEventManager

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#4266]
I was able to reproduce this with enabling the user location in `MainActivity`.

```java
app E / AndroidRuntime: FATAL EXCEPTION: main
Process: com.mapbox.mapboxsdk.testapp, PID: 1109
java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.mapboxsdk.telemetry.MapboxEventManager.pushEvent(java.util.Hashtable)'
on a null object reference
at com.mapbox.mapboxsdk.maps.MapView.trackGestureEvent(MapView.java: 1371)
at com.mapbox.mapboxsdk.maps.MapView.access$1900(MapView.java: 111)
at com.mapbox.mapboxsdk.maps.MapView$RotateGestureListener.onRotateBegin(MapView.java: 1774)
at com.almeros.android.multitouch.gesturedetectors.RotateGestureDetector.handleStartProgressEvent(RotateGestureDetector.java: 97)
at com.almeros.android.multitouch.gesturedetectors.BaseGestureDetector.onTouchEvent(BaseGestureDetector.java: 71)
at com.mapbox.mapboxsdk.maps.MapView.onTouchEvent(MapView.java: 1407)
at android.view.View.dispatchTouchEvent(View.java: 9294)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2547)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2240)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java: 2553)
at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java: 2254)
at com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java: 2403)
at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java: 1737)
at android.app.Activity.dispatchTouchEvent(Activity.java: 2765)
at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java: 60)
at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java: 60)
at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java: 2364)
at android.view.View.dispatchPointerEvent(View.java: 9514)
at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java: 4230)
at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java: 4096)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java: 3642)
at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java: 3695)
at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java: 3661)
at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java: 3787)
at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java: 3669)
at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java: 3844)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java: 3642)
at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java: 3695)
at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java: 3661)
at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java: 3669)
at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java: 3642)
at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java: 5922)
at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl
```

**Comment by tobrun** [comment]
The way we currently reference `MapboxMapEventManager` is not clean. Going to refactor

**Comment by tobrun** [comment]
Merged with #4267
